### ISSUE TITLE
useTransition on Typescript

### ISSUE THREAD

**Issue by LOGANLEEE** [pmndrs/react-spring#1396]
## 🐛 Bug Report

(A clear and concise description of what the bug is.)

## To Reproduce

`export const Profile: FunctionComponent<Props & HTMLAttributes<HTMLDivElement>> = ({ className }) => { 
const [toggle, set] = useState(false);
	const transitions = useTransition(toggle, null, {
		from: { position: 'absolute', opacity: 0 },
		enter: { opacity: 1 },
		leave: { opacity: 0 },
	});

	return (
		<Wrapper className={className}>
			{transitions.map(({ item, key, props }) =>
				item ? <animated.div style={props}>😄</animated.div> : <animated.div style={props}>🤪</animated.div>,
			)}
		</Wrapper>
	);
};`

got this error 
`
Argument of type 'null' is not assignable to parameter of type '{ ref?: SpringRef<Lookup<any>> | undefined; from?: TransitionFrom<boolean>; loop?: LoopProp<ControllerUpdate<Lookup<any>>> | undefined; ... 23 more ...; onDestroyed?: ((item: boolean, key: Key) => void) | undefined; } | (object & {})'.  TS2345

    29 | 
    30 |        const [toggle, set] = useState(false);
  > 31 |        const transitions = useTransition(toggle, null, {
       |                                                  ^
`

if I replace as {}

then I got this error

`Argument of type '{ from: { position: string; opacity: number; }; enter: { opacity: number; }; leave: { opacity: number; }; }' is not assignable to parameter of type 'any[]'.
  Object literal may only specify known properties, and 'from' does not exist in type 'any[]'.  TS2345

    33 |                {},
    34 |                {
  > 35 |                        from: { position: 'absolute', opacity: 0 },
       |                        ^
`

## Expected behavior

(A clear and concise description of what you expected to happen.)

## Link to repro (highly encouraged)

(Please provide either a [CodeSandbox demo](https://codesandbox.io) or an example GitHub repo.)

## Environment

- "react-spring": "^9.0.0",
- "react": "^17.0.2",

**Comment by joshuaellis** [comment]
This issue is badly formatted and you've missed a lot of the template out so I'm afraid this needs to be sorted before we can address it.

**Comment by alii** [comment]
I am able to reproduce this unfortunately
![image](https://user-images.githubusercontent.com/25351731/113686374-19c20e00-96bf-11eb-9311-424ef5e12bfb.png)

**Comment by joshuaellis** [comment]
Second argument of `useTransition` should be props, you no longer need to provide keys. See https://aleclarson.github.io/react-spring/v9/breaking-changes/#The-useTransition-hook

**Comment by alii** [comment]
The docs are outdated on the landing page: https://www.react-spring.io/docs/hooks/use-transition

Should I make a new issue for this?

**Comment by joshuaellis** [comment]
@alii, no please dont. We're aware of this and are working on the docs see https://github.com/pmndrs/react-spring.io/pull/49

However, if you want to to contribute to get the docs update quicker, please feel free to 😄

**Comment by joshuaellis** [comment]
I've actually pinned an issue briefly explaining the issue here #1398
